src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -23,22 +22,26 @@ internal class AttributeCloner<TAttribute>
         private readonly ConstructorInfo _bestCtor;
 
         // Compute the arguments to pass to the chosen constructor. Arguments are based on binding data.
-        private readonly Func<IReadOnlyDictionary<string, object>, object>[] _bestCtorArgBuilder;
+        private readonly Func<BindingContext, object>[] _bestCtorArgBuilder;
 
         // Compute the values to apply to Settable properties on newly created attribute.
-        private readonly Action<TAttribute, IReadOnlyDictionary<string, object>>[] _setProperties;
+        private readonly Action<TAttribute, BindingContext>[] _setProperties;
 
         // Optional hook for post-processing the attribute. This is intended for legacy hack rules.
         private readonly Func<TAttribute, Task<TAttribute>> _hook;
 
+        private IResolutionPolicy _defaultResolutionPolicy;
+
         public AttributeCloner(
             TAttribute source,
             IReadOnlyDictionary<string, Type> bindingDataContract,
             INameResolver nameResolver = null,
-            Func<TAttribute, Task<TAttribute>> hook = null)
+            Func<TAttribute, Task<TAttribute>> hook = null,
+            IResolutionPolicy defaultResolutionPolicy = null)
```

Comment:
>IResolutionPolicy defaultResolutionPolicy = null [](start = 12, length = 48)

This feels wrong:
1. attribute wide?  What happens when different properties have different resolution policies. 
2. defaulting to null?  Why can't we just grab it off the [AutoResolve] attribute?